packages/container/lib/container.js

```diff
@@ -182,6 +182,10 @@ Container.prototype = {
   }
 })();
 
+function isSingleton(container, fullName) {
+  return container._registry.getOption(fullName, 'singleton') !== false;
```

Comment:
or even `return container._registry.getOption(fullName, 'singleton')` if I'm reading this code correctly.